### ISSUE TITLE
While configuring FluxCD inside Kubernetes, Is it possible to configure persistent volume claim instead of tmp directory?

### ISSUE THREAD

**Issue by rajagopalrao** [fluxcd/community#389]
We have configured FluxCD on on premises Kubernetes cluster (on Linux servers). Recently we started noticing that tmp root directory  (5 Gi) is running out of space When scheduling new workloads getting an out of space error. 

So, we would like to know if there is any possibility configure external storage (portworx pvc) instead of tmp root directory while configuring FluxCD. Looking forward for your reply. 

Below is the error message.

**When scheduling new workloads getting an out of space error:
failed to copy temp contents: write /tmp/fetch.751668774.tmp: no space left on device**

**Comment by stefanprodan** [comment]
Yes you can use PVCs for all Flux controller volumes, here is an example https://fluxcd.io/flux/installation/configuration/vertical-scaling/#persistent-storage-for-flux-internal-artifacts

I guess the tmp issue comes from the fact that you've create a GitRepository that points to some app repo that has many GB of data. I suggest you use a dedicated repo for the manifests that should take under a Mb for all YAMLs.
